lib/parallel_tests/test/runner.rb

```diff
@@ -99,7 +132,26 @@ def execute_command(cmd, process_number, num_processes, options)
 
           print_command(cmd, env) if report_process_command?(options) && !options[:serialize_stdout]
 
-          execute_command_and_capture_output(env, cmd, options)
+          result = []
+          process_in_batches(cmd, 8191, options[:files].first).each do |subcmd|
+            result << execute_command_and_capture_output(env, subcmd, options)
+          end
+
+          # combine the output of result array into a single Hash
+          combined_result = {}
+          result.each do |res|
+            if combined_result.empty?
+              combined_result = res
+            else
+              combined_result[:env] = res[:env]
+              combined_result[:stdout] = combined_result[:stdout].to_s + res[:stdout].to_s
+              combined_result[:exit_status] = combined_result[:exit_status] + res[:exit_status] # just add
+              combined_result[:command] = combined_result[:command] | res[:command]
```

Comment:
maybe this ?
```
combined_result[:command] = cmd
```

Reply:
can be. but combined from actual res[:command] only makes it more authentic on what was executed .. i.e. nothing was missed during batch split